<launch>
	<arg name="sens" default="0.2"/>

	<!-- Start Optitrack data streaming node -->
	<include file="$(find mocap_optitrack)/launch/mocap.launch"/>

	<!-- Start Joy node: user control with joystick -->
	<node pkg="joy" type="joy_node" name="JoyInput">
		<!-- REPLACE value WITH THE RIGHT JOYSTICK IDENTIFIERˇˇˇ-->
		<param name="dev" type="string" value="/dev/input/js1"/>
		<param name="deadzone" value="0"/>
	</node>

	<!-- Start all Sphero related nodes in their namespaces-->
	<group ns="sphero_0">
		<node pkg="sphero_formation" type="man_control.py" name="manual" output="screen">
			<remap from="/joystick_input" to="/joy"/>
			<param name="sensitivity" value="$(arg sens)"/>
		</node>

        <node pkg="sphero_formation" type="kalman_filter.py" name="kalman" output="screen"/>
	</group>

	<group ns="sphero_1">
		<node pkg="sphero_formation" type="man_control.py" name="manual">
			<remap from="/joystick_input" to="/joy"/>
			<param name="sensitivity" value="$(arg sens)"/>
		</node>

        <node pkg="sphero_formation" type="kalman_filter.py" name="kalman"/>
	</group>

	<group ns="sphero_2">
		<node pkg="sphero_formation" type="man_control.py" name="manual">
			<remap from="/joystick_input" to="/joy"/>
			<param name="sensitivity" value="$(arg sens)"/>
		</node>

        <node pkg="sphero_formation" type="kalman_filter.py" name="kalman"/>
	</group>

	<group ns="sphero_3">
		<node pkg="sphero_formation" type="man_control.py" name="manual">
			<remap from="/joystick_input" to="/joy"/>
			<param name="sensitivity" value="$(arg sens)"/>
		</node>

        <node pkg="sphero_formation" type="kalman_filter.py" name="kalman"/>
	</group>
</launch>